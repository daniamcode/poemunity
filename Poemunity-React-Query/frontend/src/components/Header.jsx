import React, { useState, useEffect, useContext } from 'react'
import { Link } from 'react-router-dom'
import './Header.scss'
import '../App.scss'
import Accordion from './SimpleAccordion'
import CircularProgress from './CircularIndeterminate'
import LoginButton from './LoginButton'
import LogoutButton from './Logout'
import { useAuth0 } from '@auth0/auth0-react'
import {
  WEB_SUBTITLE,
} from '../data/constants'
import { AppContext } from '../App';
import {useLocation} from 'react-router-dom'

function Header () {
  const { isAuthenticated, isLoading } = useAuth0()
  const context = useContext(AppContext);
  const location = useLocation()

  useEffect(()=>{
    const loggedUserJSON = window.localStorage.getItem('loggedUser')
    context.setState({ user: JSON.parse(loggedUserJSON) })
  }, [JSON.stringify(location)])
  
  if (isLoading) {
    return <CircularProgress />
  }
  return (
    <>
      <section className='header'>
        <div className='header__dropdown'>
          <Accordion />
        </div>
        <div className='header__logo'>
          <Link to='/' className='header__text-logo-first'>
            P
          </Link>
          <Link to='/' className='header__logo-icon' />
          <Link to='/' className='header__text-logo-second'>
            emunity
          </Link>
        </div>
        <p className='list__presentation'>{WEB_SUBTITLE}</p>
        <div className='separator' />
        {context.user ? (
          <Link to='/profile' className='header__profile' />
        ) : (
          <></>
        )}
        {context.user ? <LogoutButton /> : <LoginButton />}
      </section>
    </>
  )
}

export default Header
