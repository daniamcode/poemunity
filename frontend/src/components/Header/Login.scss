.login-container {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;
    height: 90vh;
    .login-card {
        display: flex;
        justify-content: center;
        align-items: center;
        width: 50%;
        height: 50%;
        border: 1px solid #eee;
        box-shadow: 12px 12px 12px #0000001a;
        border-radius: 8px;
        margin-bottom: 40px;
        .login {
            display: flex;
            flex-direction: column;
            justify-content: center;
            align-items: center;
            width: 100%;
            height: 100%;
            label {
                font-size: 20px;
                padding-bottom: 20px;
            }
            .login__username,
            .login__password {
                padding: 7px 0;
                input {
                    font-size: 20px;
                    width: 20rem;
                    height: 3rem;
                }
            }
            button {
                font-size: 20px;
                width: 25rem;
                height: 3rem;
                margin: 20px;
            }
        }
    }
}

