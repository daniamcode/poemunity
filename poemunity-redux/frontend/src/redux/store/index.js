import { applyMiddleware, createStore } from 'redux';
import { createLogger }                 from 'redux-logger';
import thunk                            from 'redux-thunk';
import { composeWithDevTools }          from 'redux-devtools-extension';

import rootReducer      from '../reducers';

const { NODE_ENV } = process.env


// eslint-disable-next-line import/no-mutable-exports
let configureStore;
if (NODE_ENV !== 'production') {
    const enhancer = composeWithDevTools(
        applyMiddleware(thunk, createLogger()),
    );

    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, enhancer);
    };
} else {
    // eslint-disable-next-line no-shadow
    configureStore = function configureStore(initialState) {
        return createStore(rootReducer, initialState, applyMiddleware(thunk));
    };
}

export default configureStore;
