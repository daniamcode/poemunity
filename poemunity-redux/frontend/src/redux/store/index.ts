import { configureStore } from '@reduxjs/toolkit'
import {rootReducer}      from '../reducers/rootReducer';
import { useDispatch } from 'react-redux'


const store = configureStore({
    reducer: rootReducer,
  })

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch // Export a hook that can be reused to resolve types
export default store


// import { applyMiddleware, createStore } from 'redux';
// import { createLogger }                 from 'redux-logger';
// import thunk                            from 'redux-thunk';
// import { composeWithDevTools }          from 'redux-devtools-extension';

// const { NODE_ENV } = process.env

// let configureStore
// if (NODE_ENV !== 'production') {
//     const enhancer = composeWithDevTools(
//         applyMiddleware(thunk, createLogger()),
//     );

//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, enhancer);
//     };
// } else {
//     configureStore = function configureStore(initialState) {
//         return createStore(rootReducer, initialState, applyMiddleware(thunk));
//     };
// }

// export default configureStore;

