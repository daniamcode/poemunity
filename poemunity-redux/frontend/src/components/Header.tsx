import { useEffect, useContext } from 'react'
import { Link } from 'react-router-dom'
import './Header.scss'
import '../App.scss'
import Accordion from './SimpleAccordion'
// import CircularProgress from './CircularIndeterminate'
import LoginButton from './LoginButton'
import LogoutButton from './Logout'
import {
  WEB_SUBTITLE,
} from '../data/constants'
import { AppContext } from '../App';
import {useLocation} from 'react-router-dom'
import parseJWT from '../utils/parseJWT'

function Header () {
  const context = useContext(AppContext);
  const location = useLocation()

  useEffect(()=>{
    const loggedUserJSON = window.localStorage.getItem('loggedUser') || ''
    if(loggedUserJSON) {
      context.setState({
        ...context, 
        user: JSON.parse(loggedUserJSON), 
        userId: parseJWT(JSON.parse(loggedUserJSON))?.id,
        username: parseJWT(JSON.parse(loggedUserJSON))?.username,
        picture: parseJWT(JSON.parse(loggedUserJSON))?.picture,
        config: {
          headers: {
            Authorization: `Bearer ${JSON.parse(loggedUserJSON)}`
        }}})
    }
  }, [JSON.stringify(location)])
  
  // if (isLoading) {
  //   return <CircularProgress />
  // }
  return (
    <section className='header'>
      <div className='header__dropdown'>
        <Accordion />
      </div>
      <div className='header__logo'>
        <Link to='/' className='header__text-logo-first'>
          P
        </Link>
        <Link to='/' className='header__logo-icon' />
        <Link to='/' className='header__text-logo-second'>
          emunity
        </Link>
      </div>
      <p className='list__presentation'>{WEB_SUBTITLE}</p>
      <div className='separator' />
      {context?.user ? (
        <Link to='/profile' className='header__profile' />
      ) : (
        <></>
      )}
      {context?.user ? <LogoutButton /> : <LoginButton />}
    </section>
  )
}

export default Header
