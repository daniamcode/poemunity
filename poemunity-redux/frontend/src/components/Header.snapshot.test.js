import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'
import { BrowserRouter } from 'react-router-dom'
import {
  QueryClient,
  QueryClientProvider,
} from "react-query"
import { Provider } from 'react-redux';
import store from '../redux/store';


function renderHeader (arg) {
  const defaultProps = {
    match: {
      params: {}
    }
  }

  const props = { ...defaultProps, ...arg }
  const queryClient = new QueryClient();

  return renderer.create(
    <Provider store={store}>
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <Header {...props} />
      </BrowserRouter>
    </QueryClientProvider>
    </Provider>
  )
}

describe('Header', () => {
  let HeaderTree

  beforeEach(async () => {
    HeaderTree = renderHeader()
  })

  test('should match without id', async () => {
    expect(HeaderTree).toMatchSnapshot()
  })
})
