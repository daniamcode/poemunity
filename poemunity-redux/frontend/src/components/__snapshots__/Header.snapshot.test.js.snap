// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match without id 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="header"
      >
        <div
          class="header__dropdown"
        >
          <div
            class="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              aria-controls="panel1a-content"
              aria-disabled="false"
              aria-expanded="false"
              class="MuiButtonBase-root MuiAccordionSummary-root"
              id="panel1a-header"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiAccordionSummary-content"
              >
                <div
                  class="header__dropdown-categories-icon"
                />
                <p
                  class="header__dropdown-categories"
                >
                  Categories
                </p>
              </div>
              <div
                aria-disabled="false"
                aria-hidden="true"
                class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-hidden"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner"
                >
                  <div
                    aria-labelledby="panel1a-header"
                    id="panel1a-content"
                    role="region"
                  >
                    <div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/anniversary"
                        >
                          Anniversary
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/friendship"
                        >
                          Friendship
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/funny"
                        >
                          Funny
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/haikus"
                        >
                          Haikus
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/hope"
                        >
                          Hope
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/life"
                        >
                          Life
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/love"
                        >
                          Love
                        </a>
                      </div>
                      <div
                        class="MuiAccordionDetails-root"
                      >
                        <a
                          class="header__dropdown-subcategories"
                          href="/sad"
                        >
                          Sad
                        </a>
                      </div>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/"
                      >
                        All
                      </a>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="header__logo"
        >
          <a
            class="header__text-logo-first"
            href="/"
          >
            P
          </a>
          <a
            class="header__logo-icon"
            href="/"
          />
          <a
            class="header__text-logo-second"
            href="/"
          >
            emunity
          </a>
        </div>
        <p
          class="list__presentation"
        >
          Your poem community!!!
        </p>
        <div
          class="separator"
        />
        <button
          class="header__login"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="header"
    >
      <div
        class="header__dropdown"
      >
        <div
          class="MuiPaper-root MuiAccordion-root accordion MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
        >
          <div
            aria-controls="panel1a-content"
            aria-disabled="false"
            aria-expanded="false"
            class="MuiButtonBase-root MuiAccordionSummary-root"
            id="panel1a-header"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content"
            >
              <div
                class="header__dropdown-categories-icon"
              />
              <p
                class="header__dropdown-categories"
              >
                Categories
              </p>
            </div>
            <div
              aria-disabled="false"
              aria-hidden="true"
              class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="panel1a-header"
                  id="panel1a-content"
                  role="region"
                >
                  <div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/anniversary"
                      >
                        Anniversary
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/friendship"
                      >
                        Friendship
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/funny"
                      >
                        Funny
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/haikus"
                      >
                        Haikus
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/hope"
                      >
                        Hope
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/life"
                      >
                        Life
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/love"
                      >
                        Love
                      </a>
                    </div>
                    <div
                      class="MuiAccordionDetails-root"
                    >
                      <a
                        class="header__dropdown-subcategories"
                        href="/sad"
                      >
                        Sad
                      </a>
                    </div>
                  </div>
                  <div
                    class="MuiAccordionDetails-root"
                  >
                    <a
                      class="header__dropdown-subcategories"
                      href="/"
                    >
                      All
                    </a>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="header__logo"
      >
        <a
          class="header__text-logo-first"
          href="/"
        >
          P
        </a>
        <a
          class="header__logo-icon"
          href="/"
        />
        <a
          class="header__text-logo-second"
          href="/"
        >
          emunity
        </a>
      </div>
      <p
        class="list__presentation"
      >
        Your poem community!!!
      </p>
      <div
        class="separator"
      />
      <button
        class="header__login"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
